server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -188,7 +192,38 @@ public <T> T getAggregator(RegistryKey<T> registryKey, ValuesSourceConfig values
         );
     }
 
+    public Set<String> getSupportedAggregations(ValuesSourceType valuesSourceType) {
+        return supportedAggsByValueSourceType.getOrDefault(valuesSourceType, Set.of());
+    }
+
     public AggregationUsageService getUsageService() {
         return usageService;
     }
+
+    private static Map<ValuesSourceType, Set<String>> buildSupportedAggregations(
```

Comment:
I don't think that all aggregations use the ValuesSourceRegistry. So not all aggregations have registered an aggregatorSupplier on a per values source type basis yet. We need to check which aggregations are not using ValuesSourceRegistry and make these aggregations use it.

For counter value source type, it looks that all supported aggregations use ValuesSourceRegistry.

Reply:
Anything that uses either `ArrayValuesSourceAggregationBuilder` or `MultiValuesSourceAggregationBuilder` isn't using the registry, because we never decided how we wanted that to work.  There are a few others; IIRC the joins (nested, parent-child) don't.  Might be more.  There are others that use it in non-obvious ways, like Top metrics and Composite.

Reply:
I think for some aggregations it is ok to not be registered. 

The only subclass of `ArrayValuesSourceAggregationBuilder` is matrix_stats and maybe we can leave this out initially? I think usage of this agg is low.

The test, geo_line and weighted avg extend from MultiValuesSourceAggregationBuilder. Not sure how to handle these aggregations. Maybe in case of geo_line the point field should determine with what value source type it gets registered?